src/main/java/com/sonyericsson/jenkins/plugins/bfa/PluginImpl.java

```diff
@@ -649,21 +649,33 @@ public KnowledgeBase getKnowledgeBase() {
 
     @Override
     public boolean configure(StaplerRequest req, JSONObject o) {
-        KnowledgeBase existingKb = this.knowledgeBase;
+        KnowledgeBase existingKb = knowledgeBase;
         req.bindJSON(this, o);
 
-        if (existingKb != null && !existingKb.equals(knowledgeBase)) {
+        if (knowledgeBase != null && !existingKb.equals(knowledgeBase)) {
+            try {
+                knowledgeBase.start();
+            } catch (Exception e) {
+                logger.log(Level.SEVERE, "Could not start new knowledge base, reverting ", e);
+                // since the knowledgebase is already overwritten via req.bindJSON, we need to
+                // restore the old if the new one can't be started
+                knowledgeBase = existingKb;
+                save();
+                return true;
+            }
             if (o.getBoolean("convertOldKb")) {
                 try {
-                    knowledgeBase.start();
                     knowledgeBase.convertFrom(existingKb);
                 } catch (Exception e) {
                     logger.log(Level.SEVERE, "Could not convert knowledge base ", e);
                 }
-                knowledgeBase.stop();
             }
+            existingKb.stop();
+        } else {
+            // Since we now overwrite the existing knowledgebase immediately, we need to put it
+            // back if it was equal to the old one, or if the new one is null.
+            knowledgeBase = existingKb;
```

Comment:
not sure if this case is valid?

if it's the same one there's no need to reset it, and if it's null isn't that because the user chose to clear it?

Reply:
My reason is that if you only save the configuration without changes to a Mongo KB, the stop and start for the cache wouldn't be run. For starting the cache, that wouldn't be a problem since it would be started as soon as you interact with the KB, but for stopping it, the updaterThread etc would be kept running right?
By putting back the saved instance, we won't have a started cache floating about.

As far as it being null, I would assume something went wrong in bindJSON. When is the actual knowledgebase set to null correctly?

Reply:
no idea really, I was thinking if someone cleared all the settings it might be null but not sure if that's how it behaves